addons/udes_stock/tests/test_stock_move.py

```diff
@@ -0,0 +1,145 @@
+            products_info=[{'product': self.fig, 'qty': 5}],
+            location_dest_id=self.test_received_location_01.id,
+            location_id=self.test_stock_location_01.id,
+            assign=True,
+        )
+        move = picking.move_lines.filtered(lambda ml: ml.product_id == self.fig)
+        move_lines = MoveLine.search([('product_id', '=', self.fig.id)])
+        # Check the move lines match before any action
+        self.assertEqual(picking.move_line_ids, move_lines)
+        # Complete pack 1 operation and mock function to stop the write
+        picking.move_line_ids.filtered(
+            lambda ml: ml.product_id == self.fig and ml.package_id == pack1
+        ).write({'qty_done': 2})
+        move._action_done()
+        # Check the move line ids attached to move is still there
+        mls = picking.move_line_ids
+        # Get the new move lines, check that the new move lines and old ones are the same
+        new_move_lines = MoveLine.search([('product_id', '=', self.fig.id)])
+        self.assertIn(mls.id, new_move_lines.mapped('id'))
+        self.assertEqual(new_move_lines.mapped('id'), move_lines.mapped('id'))
+
+    def test05_unreserve_initial_demand_without_write(self):
+        """ Test for _unreserve_initial_demand, without write """
+        Move = self.env['stock.move']
+        MoveLine = self.env['stock.move.line']
+        pack1 = self.create_package()
+        pack2 = self.create_package()
+        self.create_quant(self.fig.id, self.test_stock_location_01.id, 2, package_id=pack1.id)
+        self.create_quant(self.fig.id, self.test_stock_location_01.id, 2, package_id=pack2.id)
+        picking = self.create_picking(
+            self.picking_type_pick,
+            products_info=[{'product': self.fig, 'qty': 5}],
+            location_dest_id=self.test_received_location_01.id,
+            location_id=self.test_stock_location_01.id,
+            assign=True,
+        )
+        move = picking.move_lines.filtered(lambda ml: ml.product_id == self.fig)
+        move_lines = MoveLine.search([('product_id', '=', self.fig.id)])
+        # Check the move lines match before any action
+        self.assertEqual(picking.move_line_ids, move_lines)
+        # Complete pack 1 operation and mock function to stop the write
+        picking.move_line_ids.filtered(
+            lambda ml: ml.product_id == self.fig and ml.package_id == pack1
+        ).write({'qty_done': 2})
+        with patch.object(Move.__class__, '_unreserve_initial_demand'):
+            move._action_done()
+            Move.__class__._unreserve_initial_demand.assert_called()
+        # Check the move line ids attached to move is still there
+        mls = picking.move_line_ids
+        # Get the new move lines
+        new_move_lines = MoveLine.search([('product_id', '=', self.fig.id)])
+        self.assertIn(mls.id, new_move_lines.mapped('id'))
+        self.assertNotEqual(new_move_lines.mapped('id'), move_lines.mapped('id'))
```

Comment:
It is essentially the same test but without the change, since it is bypassing `_unreserve_initial_demand`, which leads to delete and create pack2 move line.

I will remove it anyway, don't think is really needed.